### PR TITLE
lilypond: update devel version to 2.21.4

### PR DESCRIPTION
**Pull request by lemzwerg**
Also update some build environment variables.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.7.5 11G63
Xcode 4.6.3 4H1503

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by ryandesign** [comment]
> Also update some build environment variables.

Are all of the changes you made compatible with both lilypond @2.20.0 and lilypond-devel @2.21.4? If so, do the changes you made change the contents of any files installed by lilypond? If so, its revision would need to be increased.

File: textproc/lilypond/Portfile [hunk 1]
```diff
@@ -163,17 +163,12 @@ if {![variant_isset mactex]} {
...
 configure.env-append    GUILE=${prefix}/bin/guile18
-configure.env-append    GUILE_CONFIG=${prefix}/bin/guile18-config
-configure.env-append    GUILE_TOOLS=${prefix}/bin/guile18-tools
+configure.env-append    GUILE_FLAVOR=guile-1.8
 
 configure.env-append    PYTHON=${configure.python}
 build.env-append        PYTHON=${configure.python}
 destroot.env-append     PYTHON=${configure.python}
 
-configure.env-append    PYTHON_CONFIG=${configure.python}-config
```
Comment (nerdling) [inline textproc/lilypond/Portfile:173]: I'm a little surprised we no longer need to manually specify MacPorts' python to guarantee a compatible version exists.

**Comment by lemzwerg** [comment]
Thanks for merging.

The env changes don't cause any change in the package.  Regarding python: What problems would you expect?

**Comment by nerdling** [comment]
> Regarding python: What problems would you expect?

If any of lilypond's python scripts are invoked so that the shebang is used to figure out which python to use, we could end up with python2 trying to execute python3 code. Similarly, if any specific python modules were needed, they'd be installed for one macports' python but we might be letting the script get launched by a different python without the modules available.

**Pull request merged**